issuuservice-lib/lib/class.issuufolder.php

```diff
@@ -52,11 +52,119 @@ class IssuuFolder extends IssuuServiceAPI
     */
     public function add($params)
     {
-        $params['action'] = 'issuu.folder.add';
+        $this->setParams($params);
+        $response = $this->curlRequest(
+            $this->getApiUrl('/stacks'),
+            $params,
+            $this->headers,
+            'POST'
+        );
+
+        if (isset($response))
+        {
+            $result['stat'] = 'ok';
+            $result['stackId'] = $response;
+            return $result;
+        }
+        else
+        {
+            return $this->returnErrorJson($response);
+        }
+    }
+
+    protected function returnSingleResult($params)
+    {
+        $stackId = $params['folderId'];
+        $this->setParams($params);
         
+        $response = $this->curlRequest(
+            $this->getApiUrl('/stacks/'.$stackId),
+            array(),
+            $this->headers
+        );
+        $response = json_decode($response, true);
+        
+        if(isset($response['id']))
+        {
+            $result['stat'] = 'ok';
+            $result['stack'] = $this->clearObjectJson($response);
+
+            return $result;
+        }
+        else
+        {
+            return $this->returnErrorJson($response);
+        }
+    }
+
+    public function getUpdateData($params = array())
+    {
         return $this->returnSingleResult($params);
     }
 
+    /**
+    *   IssuuBookmark::stackList()
+    *
+    *   Lista stacks
+    *
+    *   @access public
+    *   @param array $params Correspondente aos parâmetros da requisição
+    */
+    public function stackList($params = array())
+    {
+        $this->setParams($params);
+
+        $response = $this->curlRequest(
+            $this->getApiUrl('/stacks'),
+            $this->params,
+            $this->headers,
+        );
+
+        $slug = $this->slug_section;
+        $response = json_decode($response, true);
+
+        if ($response['results'])
+        {
+            $result['stat'] = 'ok';
+            $result['totalCount'] = (int) $response['count'];
+            $result['page'] = (int) $params['page'];
+            $result['size'] = (int) $response['pageSize'];
+            $result['more'] = !!$response['links']['next'] ? true : false;
```

Comment:
Use `isset` to avoid undefined keys warnings

Reply:
Done!